Source/WebKit/Platform/Logging.cpp

```diff
@@ -27,8 +27,22 @@
 #include "config.h"
 #include "Logging.h"
 
+#include <wtf/NeverDestroyed.h>
+
 #if !LOG_DISABLED || !RELEASE_LOG_DISABLED
 
+#if ENABLE(LOGD_BLOCKING_IN_WEBCONTENT)
+namespace WebKit {
+
+RefPtr<WebKitLogClient>& webkitLogClient()
```

Comment:
Why do we need `webkitLogClient()`, why can't we use `webcoreLogClient()` everywhere, including in WebKit?

Reply:
That's a good point. The reason for this approach is that `WebCoreLogClient` only has the virtual functions for the WebCore logs, generated from `WebCoreLogMessages.in`. The class 'WebKitLogClient' has the virtual functions for the WebKit logs, generated from 'WebKitLogMessages.in', as well as the virtual functions from `WebCoreLogClient`, since it inherits from that class. I chose to have separate `.in` files in WebKit and WebCore, because I think it would be good to avoid that WebCore has information about WebKit logs.